### DISCUSSION TITLE
Will pivpn work in my case?

### DISCUSSION THREAD

**Discussion by gilgongo** [pivpn/pivpn#1725] [Q&A] [answered]
I've got the following setup on my home network:

1. Machine A running Ubuntu 22.04 with a pi-hole in DHCP mode, blocking ads for clients on the LAN. Also runs an OpenVPN client tunnelling all outbound traffic (DNS requests, etc.) though AirVPN.
2. Machine B (also Ubuntu 22.04) with incoming ssh and rsync ports open to the Internet.

If I wanted to block ads on my laptop or phone when outside the house, could I put pivpn onto Machine B without extra configuration (other than opening up the relevant port on my router)?

Thanks for any tips!

**Comment by coolapso** [comment, marked as answer]
yes!

Just make sure that during the DNS configuration step, you use a local dns and set it up to your machine A ip address
